src/app/manifest/manifest-service-json.ts

```diff
@@ -199,12 +199,31 @@ class ManifestService {
     store.dispatch(loadingStart(t('Manifest.Download')));
     try {
       const manifest = {};
+      // Adding a cache buster to work around bad cached CloudFlare data: https://github.com/DestinyItemManager/DIM/issues/5101
+      // try canonical component URL which should likely be already cached,
+      // then fall back to appending "?dim" then "?dim-[random numbers]",
+      // in case cloudflare has inappropriately cached another domain's CORS headers or a 404 that's no longer a 404
+      const cacheBusterStrings = [
+        '',
+        '?dim',
+        `?dim-${Math.random().toString().split('.')[1] ?? 'dimCacheBust'}`,
+      ];
       const futures = tableWhitelist
         .map((t) => `Destiny${t}Definition`)
         .map(async (table) => {
-          // Adding a cache buster "?dim" to work around bad cached CloudFlare data: https://github.com/DestinyItemManager/DIM/issues/5101
-          const response = await fetch(`https://www.bungie.net${components[table]}?dim`);
-          const body = await (response.ok ? response.json() : Promise.reject(response));
+          let response: Response | null = null;
+          const error: any[] = [];
+
+          for (const query of cacheBusterStrings) {
+            try {
+              response = await fetch(`https://www.bungie.net${components[table]}${query}`);
+              if (response.ok) break;
```

Comment:
My main concern here is that this can fail for things like no network (status -1). It'd be nice to see if we can narrow this down a bit, though I don't know how we can distinguish between "no network" and "blocked by CORS".

I suppose if there's really no network we'll fail all three pretty quickly.

Reply:
not sure how "no network" could occur in between fetching manifest paths and fetching JSONs
![image](https://user-images.githubusercontent.com/31990469/83809375-2de64a80-a66b-11ea-9818-8686cfa2bbeb.png)
the above code notices the missing internet before Platform/Destiny2/Manifest
fails due to 
![image](https://user-images.githubusercontent.com/31990469/83809522-6be36e80-a66b-11ea-90a2-dd1ded69e9c2.png)
then it tries to grab profile anyway, with a ton of components
![image](https://user-images.githubusercontent.com/31990469/83809596-8e758780-a66b-11ea-8407-02184408ccec.png)
which is a little confusing since not like we can do a *ton* without manifest

anyway CORS error throws a `TypeError: Failed to fetch` and pointing it at a nonexistent domain for the JSON db gives a `TypeError: Failed to fetch` but again i have no way to unplug my internet in the time between Platform/Destiny2/Manifest and getting the JSONs, and i don't think it's worth accounting for